src/System.Text.Http/System/Text/Http/Obsolete/HttpRequestParser.cs

```diff
@@ -188,26 +188,24 @@ static class HttpRequestParser
         static Utf8String s_Delete => new Utf8String(b_Delete);
 
 
-        internal static bool TryParseRequestLine(ReadOnlySpan<byte> buffer, out HttpRequestLine requestLine, out int totalParsedBytes)
```

Comment:
Here we have a form ` bool TryParse(..., out Span, out int) `
It easily gets in trouble when we pass a returnable span (which we actually intend to return) and also a ref to an ordinary int local. Callee may stash the ref into the span and that would be bad, so this triggers an error.

I also noticed that in all cases returning `false` is matched with bytes being 0, so I just changed this to a form

` int TryParse(..., out Span) ` where returning 0 indicates a failure.

Reply:
How can the ref be stashed into the span? The span is of bytes. If you stash anything into it, it will just copy the byte from the ref location.

Also, what's the error's message? If we have to keep the error, we should make sure it explains what's going on.

Reply:
Copying the bytes is not a problem. Creating a one-element span that wraps the given reference is a problem though. 
We do not know what happens inside TryParseRequestLine and must assume the worst case scenario. The worst case scenario is that TryParseRequestLine does:

```C#
bool TryParseRequestLine(ReadOnlySpan<byte> buffer, out HttpRequestLine requestLine, out int totalParsedBytes)
{
    totalParsedBytes = 42;
    requestLine = WrapInOneElementSpan(ref totalParsedBytes);
    return true;
}
```
As a result if you pass a returnable `out requestLine` together with a local-referring `out totalParsedBytes` you get an error about mixing references that have different escape scopes.

If an error is not given, you may end up returning `requestLine` that refers to stack memory, which is basically a GC hole.

The errors could be improved, but it is correct to give an error here. 
Ref-like types are considered as containing references and when they are ref/out parameters they are also assignable by the callee.
As a result mixing a ref/out ref-like argument with references that have narrower life time in the same invocation is the same as assigning one into another and that is illegal.